dolphin/api/v1/storages.py

```diff
@@ -139,22 +140,22 @@ def sync(self, req, id):
         :return:
         """
         # validate the id
-        context = req.environ.get('dolphin.context')
+        ctxt = req.environ['dolphin.context']
         # admin_context = context.RequestContext('admin', 'fake', True)
         try:
-            device = db.access_info_get(context, id)
+            storage = db.storage_get(ctxt, id)
         except Exception as e:
             LOG.error(e)
-            raise exception.AccessInfoNotFound(e)
+            raise exc.HTTPBadRequest(explanation=e.message)
 
-        tasks = (
-            'pool_task',
-            'volume_task'
-        )
-        for task in tasks:
-            self.task_rpcapi.sync_storage_resource(context, id, task)
+        for subclass in task.StorageResourceTask.__subclasses__():
+            self.task_rpcapi.sync_storage_resource(
+                ctxt,
+                id,  # its storage_id
```

Comment:
Is it better to use storage_id instead of id, so the code is self-explanatory.

Reply:
done